File: spec/java/src/io/kaitai/struct/testformats/CustomFxNoArgs.java
package io.kaitai.struct.testformats;

import io.kaitai.struct.CustomDecoder;

public class CustomFxNoArgs implements CustomDecoder {
    public CustomFxNoArgs() {
    }

    @Override
    public byte[] decode(byte[] src) {
        byte[] dst = new byte[src.length + 2];
        dst[0] = '_';
        dst[src.length + 1] = '_';
        System.arraycopy(src, 0, dst, 1, src.length);
        return dst;
    }
}

File: helpers/javascript/CustomFxNoArgs.js
'use strict';

(function (root, factory) {
  if (typeof define === 'function' && define.amd) {
    define(['exports', 'kaitai-struct/KaitaiStream'], factory);
  } else if (typeof exports === 'object' && exports !== null && typeof exports.nodeType !== 'number') {
    factory(exports, require('kaitai-struct/KaitaiStream'));
  } else {
    factory(root.CustomFxNoArgs || (root.CustomFxNoArgs = {}), root.KaitaiStream);
  }
})(typeof self !== 'undefined' ? self : this, function (CustomFxNoArgs_, KaitaiStream) {
var CustomFxNoArgs = (function() {
  function CustomFxNoArgs() {
  }
  CustomFxNoArgs.prototype.decode = function(src) {
    var len = src.length;
    var dest = new Uint8Array(len + 2);
    for (var i = 0; i < len; i++) {
      dest[i + 1] = src[i];
    }
    dest[0] = 95;
    dest[len + 1] = 95;
    return dest;
  }

  return CustomFxNoArgs;
})();
CustomFxNoArgs_.CustomFxNoArgs = CustomFxNoArgs;
});
